optexp/datasets/wikitext103.py

```diff
@@ -0,0 +1,172 @@
+import os
+import shutil
+from pathlib import Path
+
+import pyarrow.parquet
+import requests
+import torch
+from attrs import frozen
+from torch.utils.data import DataLoader
+
+from optexp.config import Config
+from optexp.datasets import Dataset
+from optexp.datasets.dataset import TrVa, HasClassCounts, Downloadable
+from optexp.datasets.tokenizers import Tokenizer, BPETokenizer
+
+
+#  TODO <unk> considered harmful
+@frozen
+class WikiText103(Dataset, HasClassCounts, Downloadable):
+
+    sequence_length: int = 1024
+    vocab_size: int = 50257
+    tokenizer: Tokenizer = BPETokenizer()
+    directory: Path = Config.get_dataset_directory() / "WikiText103"
+
+    def get_dataloader(self, b: int, tr_va: TrVa, num_workers: int) -> DataLoader:
+        dataset = self.get_dataset(tr_va)
+
+        def collate_fn(batch):
+            src_batch, tgt_batch = [], []
+            for sample in batch:
+                src_batch.append(sample[0])
+                tgt_batch.append(sample[1])
+
+            return torch.vstack(src_batch), torch.vstack(tgt_batch).reshape(-1)
```

Comment:
`batch` might benefit from a type hint. I assume it's a `List[Tuple[?,?]]`

Reply:
added